### PR TITLE
upgrade opentok ios sdk version and remove custom audio driver

### PR DESCRIPTION
**Pull request by iujielim**

### CODE REVIEW

File: iOS-Sample/Hello-World/OTNetworkTest.m [hunk 1]
```diff
@@ -184,6 +173,7 @@ - (void)doSubscribe:(OTStream*)stream
         [self dispatchResultsToDelegateWithResult:OTNetworkTestResultNotGood
                                             error:error];
     }
+    _subscriber.audioVolume = 0;
```
Comment (IGitGotIt) [inline iOS-Sample/Hello-World/OTNetworkTest.m:176]: is this a substitute for `[_myAudioDevice setAudioPlayoutMute:YES];` ?

**Comment by abdulajet** [comment]
image

image

I seem to get this error when i try to build the app for the simulator?

**Comment by iujielim** [comment]
@abdulajet Sorry about that, I forgot to remove my identifier setting in .project.pbxproj
Please try again , I tested and it is working now :)

**Review by IGitGotIt — COMMENTED** [review]
Left a comment about subscriber volume before...
thx

**Comment by IGitGotIt** [comment]
> image
> 
> image
> 
> I seem to get this error when i try to build the app for the simulator?

for me the following setting  worked
image

File: iOS-Sample/Hello-World/OTNetworkTest.m (hunk 1 above)
Comment (IGitGotIt) [reply iOS-Sample/Hello-World/OTNetworkTest.m:176]: also a  follow up question - what is the main reason to remove the default audio file ?

Comment (iujielim) [reply iOS-Sample/Hello-World/OTNetworkTest.m:176]: yes, you are right, this is the substitute of default audio file :)

**Comment by iujielim** [comment]
@IGitGotIt , can i remove VALID_ARCHS? it was there in the main branch

File: iOS-Sample/Hello-World/OTNetworkTest.m (hunk 1 above)
Comment (IGitGotIt) [reply iOS-Sample/Hello-World/OTNetworkTest.m:176]: Just out of curiosity - what is the reason to remove the custom audio file ?

Also the old way was not reading any samples and rendering - hence it was called "mute". In this new mode, by making the volume 0 you would still be reading audio frames, just not playing them out loud. I am not sure why it was done , but just to keep the difference in mind.

Personally , I do the network test with both audio and video on , to get a true picture. 

Hence the goal of what we want to accomplish is more important.

**Comment by IGitGotIt** [comment]
> @IGitGotIt , can i remove VALID_ARCHS? it was there in the main branch

I think you can, just do a quick check that a device and simulator both work, before you do it.
Thanks

**Comment by iujielim** [comment]
@IGitGotIt , I had tested and removed the VALID_ARCH in this PR.

File: iOS-Sample/Hello-World/OTNetworkTest.m (hunk 1 above)
Comment (iujielim) [reply iOS-Sample/Hello-World/OTNetworkTest.m:176]: As my understanding, we need to render audio for the audio testing in the program
In the old way we introduce a custom audio device (_myAudioDevice and default audio file), the purpose is to remove the echos while user test the microphone. Means the audio is still rendering and receive as audio input, but it is set to "playout" mute so the user wont hear their microphone echos

In new way, which was introduce in iOS SDK version 2.22.2, we can now set the playout audio to 0, so we dont need the custom audio device to help us set the playout to mute


In the debug log, you can find that the audio and video bytes is being rendered and calculate:
2022-06-01 17:55:04.872088+0800 Hello-World[13223:5389451] **audioBytesReceived** 43362, bps 38223, packetsLost 0.00
2022-06-01 17:55:04.872484+0800 Hello-World[13223:5389451] **videoBytesReceived** 713245, bps 720141, packetsLost 0.00
2022-06-01 17:55:05.708332+0800

**Comment by abdulajet** [comment]
OK building for me now, lets change the deployment target to the version of iOS OT supports now 🚀.

EDIT: Seems that the SDK still supports iOS 9, I think the project generates a bunch of warnings for using below iOS 11. So maybe bump to iOS 11?

**Comment by iujielim** [comment]
Thanks for reminding me of updating the deployment target ~ I updated the PR from 9 to 11, tested and is working fine ~

**Review by abdulajet — APPROVED** [review]
lgtm (assuming @IGitGotIt is happy)

**Review by IGitGotIt — APPROVED** [review]
+1

**Pull request merged**